core/store/src/main/java/mil/nga/giat/geowave/core/store/util/DataStoreUtils.java

```diff
@@ -703,67 +671,64 @@ else if ((vis2 == null) || (vis2.length == 0)) {
 					new byte[] {}));
 		}
 		else {
-			final Set<Integer> skipSet = new HashSet<Integer>();
-			for (final NumericDimensionField<? extends CommonIndexValue> f : indexModel.getDimensions()) {
-				if (!fieldIds.add(f.getFieldId())) {
-					// if the field ID is repeated, make sure we don't repeat
-					// the
-					// field position when reading the values
-					skipSet.add(i);
-				}
-				i++;
-			}
-			i = 0;
+			// Get the list of valid field positions from the row's field mask
+			List<Integer> fieldPositions = BitmaskUtils.getFieldPositions(row.getFieldMask());
+
+			// Collect the valid fields
+			int fieldIndex = 0;
```

Comment:
BaseDataStore's decodeRow is now sync'd up w/ HBase logic; however, we won't be able to completely integrate until we solve visibility handling.